Core/GDCore/IDE/Events/UsedExtensionsFinder.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef GDCORE_USED_EXTENSIONS_FINDER_H
+#define GDCORE_USED_EXTENSIONS_FINDER_H
+#include <set>
+
+#include "GDCore/Events/Parsers/ExpressionParser2NodeWorker.h"
+#include "GDCore/IDE/Events/ArbitraryEventsWorker.h"
+#include "GDCore/IDE/Project/ArbitraryObjectsWorker.h"
+#include "GDCore/String.h"
+
+namespace gd {
+class Project;
+class Object;
+class BehaviorContent;
+}  // namespace gd
+
+namespace gd {
+
+class GD_CORE_API UsedExtensionsFinder
+    : public ArbitraryObjectsWorker,
+      public ArbitraryEventsWorkerWithContext,
+      public ExpressionParser2NodeWorker {
+ public:
+  static std::set<gd::String> ScanProject(gd::Project& project);
+
+ private:
+  UsedExtensionsFinder(gd::Project& _project) { projectPtr = &_project; };
+  gd::Project* projectPtr;
+  std::set<gd::String> usedExtensions;
+
+  // Object Visitor
+  void DoVisitObject(gd::Object& object);
+
+  // Behavior Visitor
+  void DoVisitBehavior(gd::BehaviorContent& behavior);
+
+  // Instructions Visitor
+  bool DoVisitInstruction(gd::Instruction& instruction, bool isCondition);
```

Comment:
nit: you can use `override` keyword to ensure you're properly overriding a virtual method. Can help for silly naming/const mistakes that are hard to debug.